src/coinbase/smart_contract.ts

```diff
@@ -1,10 +1,220 @@
+   */
+  public static fromModel(contractModel: SmartContractModel): SmartContract {
+    return new SmartContract(contractModel);
+  }
+
+  /**
+   * Returns the ID of the SmartContract.
+   *
+   * @returns The SmartContract ID.
+   */
+  public getId(): string {
+    return this.model.smart_contract_id;
+  }
+
+  /**
+   * Returns the Network ID of the SmartContract.
+   *
+   * @returns The Network ID.
+   */
+  public getNetworkId(): string {
+    return this.model.network_id;
+  }
+
+  /**
+   * Returns the Wallet ID that deployed the smart contract.
+   *
+   * @returns The Wallet ID.
+   */
+  public getWalletId(): string {
+    return this.model.wallet_id;
+  }
+
+  /**
+   * Returns the Contract Address of the smart contract.
+   *
+   * @returns The Contract Address.
+   */
+  public getContractAddress(): string {
+    return this.model.contract_address;
+  }
+
+  /**
+   * Returns the Deployer Address of the smart contract.
+   *
+   * @returns The Deployer Address.
+   */
+  public getDeployerAddress(): string {
+    return this.model.deployer_address;
+  }
+
+  /**
+   * Returns the Type of the smart contract.
+   *
+   * @returns The Smart Contract Type.
+   */
+  public getType(): SmartContractType {
+    switch (this.model.type) {
+      case SmartContractTypeModel.Erc20:
+        return SmartContractType.ERC20;
+      case SmartContractTypeModel.Erc721:
+        return SmartContractType.ERC721;
+      default:
+        throw new Error(`Unknown smart contract type: ${this.model.type}`);
+    }
+  }
+
+  /**
+   * Returns the Options of the smart contract.
+   *
+   * @returns The Smart Contract Options.
+   */
+  public getOptions(): SmartContractOptions {
+    if (this.getType() === SmartContractType.ERC20) {
+      return this.model.options as TokenContractOptions;
+    }
+    return this.model.options as NFTContractOptions;
+  }
+
+  /**
+   * Returns the ABI of the smart contract.
+   *
+   * @returns The ABI as a JSON-encoded string.
+   */
+  public getAbi(): string {
+    return this.model.abi;
+  }
+
+  /**
+   * Returns the Transaction of the smart contract deployment.
+   *
+   * @returns The Transaction.
+   */
+  public getTransaction(): Transaction {
+    return new Transaction(this.model.transaction);
+  }
+
+  /**
+   * Signs the SmartContract deployment with the provided key and returns the hex signature
+   * required for broadcasting the SmartContract deployment.
+   *
+   * @param key - The key to sign the SmartContract deployment with
+   * @returns The hex-encoded signed payload
+   */
+  async sign(key: ethers.Wallet): Promise<string> {
+    return this.getTransaction().sign(key);
+  }
+
+  /**
+   * Broadcasts the SmartContract deployment to the Network.
+   *
+   * @returns The SmartContract object
+   * @throws {APIError} if the API request to broadcast a SmartContract deployment fails.
+   */
+  public async broadcast(): Promise<SmartContract> {
+    if (!this.getTransaction()?.isSigned())
```

Comment:
Is the `?` necessary if the typing of `getTransaction` is not nullable?

Reply:
fixed!